.github/workflows/mypy.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow will install Python dependencies and run MyPy
+
+name: MyPy Type Checking
+
+on: [pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      id: setup_python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.10"
+    - name: Restore Virtualenv
+      uses: actions/cache/restore@v4
+      id: cache-venv-restore
+      with:
+        path: ./.venv/
+        key: ${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-venv-${{ hashFiles('*requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-venv-
+    - name: Install dependencies
+      run: |
+        # Create the virtual environment
+        python -m venv .venv
+        . ./.venv/bin/activate
+
+        # Install the dependencies
+        # In case of a cache hit on the primary key, this will be a no-op
+        # In case of a cache miss, but hit on a secondary key, this will update what's changed
+        python -m pip install --upgrade pip
+        pip install -r test-requirements.txt
```

Comment:
I'm not exactly sure how `save` works but we should make sure that we don't save the install of fms. we want to be able to make changes on the main branch of fms and have this repo immediately see them.

Reply:
Good point, I should exclude the install of `fms` from the cache, thanks